src/System.Collections/src/System/Collections/Generic/SortedSet.cs

```diff
@@ -1132,13 +1132,14 @@ internal virtual void IntersectWithEnumerable(IEnumerable<T> other)
                 if (Contains(item))
                 {
                     toSave.Add(item);
+                    DoRemove(item);
                 }
             }
 
-            if (toSave.Count < Count)
+            Clear();
+            foreach (T item in toSave)
             {
-                Clear();
-                AddAllElements(toSave);
+                AddIfNotPresent(item);
```

Comment:
Ah, answered my own question:
https://github.com/dotnet/corefx/blob/4d54861a9cc3203f5a590f13a8ce0e0e4ebebd02/src/System.Collections/src/System/Collections/Generic/SortedSet.cs#L338
That seems... odd... but ok.